sklearn/decomposition/nmf.py

```diff
@@ -402,12 +402,13 @@ def _init(self, X):
             W, H = _initialize_nmf(X, self.n_components_, variant='ar',
                                    random_state=rng)
         elif init == "random":
-            W = rng.randn(n_samples, self.n_components_)
+            scale = np.sqrt(X.std()) / self.n_components_ ** (1. / 4)
```

Comment:
That was the plan.
I also made some changes in #4852, but my scaling is different:
I use `np.sqrt(X.mean() / self.n_components_)`, so the product W.H is close to X in mean, but it may needs more investigations.